pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -498,16 +483,11 @@ func handleDelete(ctx *downloaderContext, key string,
 		log.Fatalf("handleDelete: No ObjType for %s",
 			status.ImageID)
 	}
-	locDirname := types.DownloadDirname + "/" + status.ObjType
+	locDirname := status.FileLocation
```

Comment:
Isn't this also a filename and not a dirname?

Reply:
Not quite. This is a bit legacy. `status.FileLocation` is the directory where the file is, not the file itself.

I think it is not used anywhere else, but I was not sure. Once this is all clean except for this, and all tests pass, then I can try to remove it and see if they still pass.